ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -1433,17 +1452,19 @@ void handleResponse(ResponseInfo responseInfo, PutResponse putResponse) {
           routerMetrics.crossColoSuccessCount.inc();
         }
       } else {
-        onErrorResponse(chunkPutRequestInfo.replicaId);
+        onErrorResponse(chunkPutRequestInfo.replicaId, routerErrorCode);
       }
       checkAndMaybeComplete();
     }
 
     /**
      * Perform the necessary actions when a request to a replica fails.
      * @param replicaId the {@link ReplicaId} associated with the failed response.
+     * @param routerErrorCode the {@link RouterErrorCode} associated with failed response.
      */
-    private void onErrorResponse(ReplicaId replicaId) {
-      operationTracker.onResponse(replicaId, TrackedRequestFinalState.FAILURE);
+    private void onErrorResponse(ReplicaId replicaId, RouterErrorCode routerErrorCode) {
```

Comment:
If the new `RouterErrorCode`,  `OperationDisabled` is only used to assign `TrackedRequestFinalState`, why not pass `TrackedRequestFinalState` into this method instead of adding a new router code?

Reply:
You're right. Will make the change.